### ISSUE TITLE
Header component reports warning that tag prop has been deprecated even if not specified

### ISSUE THREAD

**Issue by tobias-flitsch** [grommet/grommet#882]
The `Header` component reports a deprecated warning even if the `tag` property is not specified. The component contains `tag` in the default properties, which could cause this warning.

### Expected Behavior
If no tag property is specified, no warning should be logged.

### Actual Behavior
A warning is displayed.

### URL, screen shot, or Codepen exhibiting the issue

![image](https://cloud.githubusercontent.com/assets/4466053/18773642/cd0e3b7a-8105-11e6-8e4e-5b74c3388b97.png)

![image](https://cloud.githubusercontent.com/assets/4466053/18773639/ba3162ca-8105-11e6-9d4a-6d036dfa7a9d.png)

```
if (this.props.tag && 'header' !== this.props.tag) {
    console.warn('Header: tag prop has been deprecated. ' + 'Use a separate Heading instead.');
}
```
```
Header.defaultProps = {
  pad: { horizontal: 'none', vertical: 'none', between: 'small' },
  direction: 'row',
  align: 'center',
  responsive: false,
  tag: 'header' // remove in 1.0
};
```

### Steps to Reproduce
 1. Create `Header` component and render.
 3. Check console output for the deprecated warning.

### Your Environment

* Grommet version: v0.6.12
* Browser Name and version: Google Chrome 53.0.2785.116 (Official Build) (64-bit)
* Operating System and version (desktop or mobile): macOS Sierra (10.12)

**Comment by bryanjacquot** [comment]
This issue will be resolved with #777

**Comment by bryanjacquot** [comment]
Fixed in https://github.com/grommet/grommet/commit/036303b51335840fe4498e67384f4ecc82aaa946 on stable branch.
